### ISSUE TITLE
NXOS - Oxidized doesn't handle incorrect password correctly

### ISSUE THREAD

**Issue by derekivey** [ytti/oxidized#2599]
It seems Oxidized gets hung up and stops updating configs if the device doesn't accept the password it tries to login with. I had a few devices where the password was not properly set on the device. When this happens, the device asks for the password again, for example:

```
User Access Verification
Password: 
Password: 
Password: 
Received disconnect from X.X.X.X port 22:2: Too many authentication failures
Authentication failed.
```

With debug mode enabled, I see the logs look like this:

```
Aug 10 11:35:31 oxidized01.local docker[23956]: oxidized[24]: lib/oxidized/worker.rb: Jobs running: 0 of 1 - ended: 25 of 34
Aug 10 11:35:31 oxidized01.local docker[23956]: oxidized[24]: lib/oxidized/worker.rb: Added /sw01 to the job queue
Aug 10 11:35:31 oxidized01.local docker[23956]: oxidized[24]: lib/oxidized/worker.rb: 1 jobs running in parallel
Aug 10 11:35:31 oxidized01.local docker[23956]: oxidized[24]: lib/oxidized/job.rb: Starting fetching process for sw01 at 2022-08-10 18:35:31 UTC
Aug 10 11:35:31 oxidized01.local docker[23956]: oxidized[24]: lib/oxidized/input/ssh.rb: Connecting to sw01
Aug 10 11:35:31 oxidized01.local docker[23956]: oxidized[24]: AUTH METHODS::["none", "password", "keyboard-interactive"]
Aug 10 11:35:32 oxidized01.local docker[23956]: oxidized[24]: lib/oxidized/worker.rb: 1 jobs running in parallel
Aug 10 11:35:33 oxidized01.local docker[23956]: oxidized[24]: lib/oxidized/worker.rb: 1 jobs running in parallel
Aug 10 11:35:34 oxidized01.local docker[23956]: oxidized[24]: lib/oxidized/worker.rb: 1 jobs running in parallel
Aug 10 11:35:35 oxidized01.local docker[23956]: oxidized[24]: lib/oxidized/worker.rb: 1 jobs running in parallel
Aug 10 11:35:36 oxidized01.local docker[23956]: Password: oxidized[24]: lib/oxidized/worker.rb: 1 jobs running in parallel
Aug 10 11:35:37 oxidized01.local docker[23956]: oxidized[24]: lib/oxidized/worker.rb: 1 jobs running in parallel
Aug 10 11:35:38 oxidized01.local docker[23956]: oxidized[24]: lib/oxidized/worker.rb: 1 jobs running in parallel
Aug 10 11:35:39 oxidized01.local docker[23956]: oxidized[24]: lib/oxidized/worker.rb: 1 jobs running in parallel
```

You can see above that "Password:" seems to get returned and the job seems to just get stuck in a loop.

**Comment by derekivey** [comment]
This is still an issue. We also hit this for devices which were missing a DNS record:

```
May 03 12:30:01 oxidized01.local docker[70790]: oxidized[24]: lib/oxidized/nodes.rb: Loading nodes
May 03 12:30:02 oxidized01.local docker[70790]: oxidized[24]: lib/oxidized/nodes.rb: Loaded 224 nodes
May 03 12:38:04 oxidized01.local docker[70790]: oxidized[24]: atl01-ce03 raised SocketError with msg "getaddrinfo: Name or service not known", atl01-ce03 saved
May 03 12:38:05 oxidized01.local docker[70790]: oxidized[24]: /atl01-ce03 status no_connection, retry attempt 1
May 03 12:40:01 oxidized01.local docker[70790]: oxidized[24]: lib/oxidized/nodes.rb: Loading nodes
May 03 12:40:02 oxidized01.local docker[70790]: oxidized[24]: lib/oxidized/nodes.rb: Loaded 224 nodes
May 03 12:46:38 oxidized01.local docker[70790]: oxidized[24]: atl01-ce03 raised SocketError with msg "getaddrinfo: Name or service not known", atl01-ce03 saved
May 03 12:46:39 oxidized01.local docker[70790]: oxidized[24]: /atl01-ce03 status no_connection, retry attempt 1
May 03 12:50:01 oxidized01.local docker[70790]: oxidized[24]: lib/oxidized/nodes.rb: Loading nodes
May 03 12:50:02 oxidized01.local docker[70790]: oxidized[24]: lib/oxidized/nodes.rb: Loaded 224 nodes
May 03 12:55:12 oxidized01.local docker[70790]: oxidized[24]: atl01-ce03 raised SocketError with msg "getaddrinfo: Name or service not known", atl01-ce03 saved
May 03 12:55:13 oxidized01.local docker[70790]: oxidized[24]: /atl01-ce03 status no_connection, retry attempt 1
```

Oxidized got hung up and stopped backing up the rest of our devices.

**Comment by derekivey** [comment]
I believe I've finally figured out a fix for this. Disabling keyboard-interactive authentication seems to do the trick.

This is done by setting (in oxidized config file):

```
vars:
  auth_methods: [ "password" ]
```

In my case we only use password authentication.

Without this change we saw connections stuck open in a CLOSE_WAIT state (netstat -an | grep CLOSE_WAIT) and with the fix we now see them in TIME_WAIT (netstat -an | grep TIME_WAIT) and go away within a few minutes.
